WordPress/Classes/Networking/TaxonomyServiceRemoteREST.m

```diff
@@ -2,76 +2,148 @@
+{
+    [self getTagsWithSuccess:success
+                      paging:nil
+                     failure:failure];
+}
+
+- (void)getTagsWithSuccess:(void (^)(NSArray<RemotePostTag *> *))success
+                    paging:(RemoteTaxonomyPaging *)paging
+                   failure:(void (^)(NSError *))failure
+{
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTTagTypeIdentifier
+                   parameters:[self parametersForPaging:paging]
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteTagsWithJSONArray:[responseObject arrayForKey:@"tags"]]);
+                          }
+                      } failure:failure];
+}
+
+- (void)searchTagsWithName:(NSString *)nameQuery
+                   success:(void (^)(NSArray <RemotePostTag *> *tags))success
+                   failure:(void (^)(NSError *error))failure
+{
+    NSParameterAssert(nameQuery.length > 0);
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTTagTypeIdentifier
+                   parameters:@{@"search": nameQuery}
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteTagsWithJSONArray:[responseObject arrayForKey:@"tags"]]);
+                          }
+                      } failure:failure];
+}
+
+#pragma mark - default methods
+
+- (void)createTaxonomyWithType:(NSString *)typeIdentifier
+                    parameters:(NSDictionary *)parameters
+                       success:(void (^)(id responseObject))success
+                       failure:(void (^)(NSError *error))failure
+{
+    NSString *path = [NSString stringWithFormat:@"sites/%@/%@/new?context=edit", self.siteID, typeIdentifier];
+    NSString *requestUrl = [self pathForEndpoint:path withVersion:ServiceRemoteRESTApiVersion_1_1];
+    
     [self.api POST:requestUrl
         parameters:parameters
-           success:^(AFHTTPRequestOperation *operation, id responseObject) {
-               RemotePostCategory *receivedCategory = [self remoteCategoryWithJSONDictionary:responseObject];
+           success:^(AFHTTPRequestOperation * _Nonnull operation, id  _Nonnull responseObject) {
                if (success) {
-                   success(receivedCategory);
+                   success(responseObject);
                }
-           } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+           } failure:^(AFHTTPRequestOperation * _Nullable operation, NSError * _Nonnull error) {
                if (failure) {
                    failure(error);
                }
            }];
 }
 
-#pragma mark - tags
-
-- (void)getTagsWithSuccess:(void (^)(NSArray <RemotePostTag *> *tags))success
-                   failure:(void (^)(NSError *error))failure
+- (void)getTaxonomyWithType:(NSString *)typeIdentifier
+                 parameters:(id)parameters
+                    success:(void (^)(id responseObject))success
```

Comment:
Do parameters and responseObject need to be `id`s? Could they be concrete types?